### PR TITLE
Add the ability to use a Tile ControlNet for iterative tiled upscaling

### PR DESCRIPTION
**Pull request by llotenberg**
I frequently use the Tile ControlNet when I am using tiled upscaling, because from my experience it makes the results much more cohesive than standard tiling without having to use an incredibly vague prompt. I find that I get the best results from using each upscaled image for the ControlNet, which means it needs to be used partway through the IterativeLatentUpscale process for the best results. As such, I decided to add support for easily using the Tile ControlNet in the TiledKSample upscaling process provided by this repo. (This functionality will require comfy-controlnet-aux to be installed in order to work correctly.)

I apologize for submitting this while my other pull request is still under review, but I also intend to work on implementing the SD-Latent-Upscaler functionality for upscaling eventually, and since it involves almost the exact same areas of code, I'd prefer to have this pull request confirmed or denied before I begin work on that (to ease later merging). I am considering allowing the pyrUp_iters of the Tile Preprocessor to be manipulated by the user, though, so I'm going to submit this as a draft for now.

### CODE REVIEW

**Comment by ltdrdata** [comment]
It is recommended to use https://github.com/Ttl/ComfyUi_NNLatentUpscale as the upscaler in the latent space. NNLatentUpscale allows for more flexible adjustment of the upscale size and does not produce the stripe pattern observed in SD-Latent-Upscaler.

And when performing Latent Upscale, it is important to keep in mind that the resolution must be in multiples of 8 in pixel space.

**Comment by llotenberg** [comment]
@ltdrdata Understood, I'll keep that in mind when working on that functionality.

On the matter of this pull request, I attempted to add the ability to modify the pyrUp_iters parameter (for the preprocessor), but couldn't get it to work correctly on the JavaScript side of things. As such, I'm simply going to leave it with a reasonable hardcoded value, and open the commit for review. (I may revisit this later on.)

**Comment by ltdrdata** [comment]
This workflow is failing...

https://github.com/ltdrdata/ComfyUI-extension-tutorials/blob/Main/ComfyUI-Impact-Pack/workflow/extreme-highscale.png

```
Traceback (most recent call last):
  File "/mnt/teratera/git/ComfyUI/execution.py", line 153, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/teratera/git/ComfyUI/execution.py", line 83, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/teratera/git/ComfyUI/execution.py", line 76, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/teratera/git/ComfyUI-Impact-Pack/modules/impact/impact_pack.py", line 1078, in doit
    current_latent = upscaler.upscale_shape(step_info, current_latent, new_w, new_h, temp_prefix)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/teratera/git/ComfyUI-Impact-Pack/modules/impact/core.py", line 1604, in upscale_shape
    refined_latent = nodes.KSampler().sample(model, seed, steps, cfg, sampler_name, scheduler,
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/teratera/git/ComfyUI/nodes.py", line 1299, in sample
    return common_ksampler(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/teratera/git/ComfyUI/nodes.py", line 1256, in common_ksampler
    latent_image = latent["samples"]
                   ~~~~~~^^^^^^^^^^^
TypeError: tuple indices must be integers or slices, not str


```

**Comment by llotenberg** [comment]
Ah, I think I accidentally broke other upscaling providers. I'll fix it
later today.

**Comment by llotenberg** [comment]
Alright, it should be fixed now. Sorry about the delay. (I didn't actually check it because I don't think my computer can't handle that workflow, though.)

File: modules/impact/core.py [hunk 1]
```diff
@@ -1639,25 +1653,39 @@ class PixelTiledKSampleUpscaler:
...
+        if self.tile_cnet is not None:
+            image_batch, image_w, image_h, _ = images.shape
+            if image_batch > 1:
+                warnings.warn('Multiple latents in batch, Tile ControlNet being ignored')
+            else:
+                if 'TilePreprocessor' not in nodes.NODE_CLASS_MAPPINGS:
+                    raise RuntimeError("'TilePreprocessor' node (from comfyui_controlnet_aux) isn't installed.")
+                preprocessor = nodes.NODE_CLASS_MAPPINGS['TilePreprocessor']()
+                # might add capacity to set pyrUp_iters later, not needed for now though
+                preprocced = preprocessor.execute(images, pyrUp_iters=3, resolution=min(image_w, image_h))
```
Comment (ltdrdata) [inline modules/impact/core.py]: I think you forgot to pass the `preprocessed` image to `apply_cnet`.

**Review by ltdrdata — CHANGES_REQUESTED** [review]

File: modules/impact/core.py (hunk 1 above)
Comment (ltdrdata) [reply modules/impact/core.py]: The review was pending due to my mistake.

Comment (llotenberg) [reply modules/impact/core.py]: D'oh, you're right. Fixed.

**Review by ltdrdata — APPROVED** [review]

**Pull request merged**